lib/activerecord-multi-tenant/model_extensions.rb

```diff
@@ -42,20 +42,15 @@ def partition_key
                                         .try(:instance_variable_get, :@partition_key)
           end
 
-          # Avoid primary_key errors when using composite primary keys (e.g. id, tenant_id)
```

Comment:
In supporting multiple versions, the variable states and values have become more complex.
I tried to override the `#reset_primary_key` private method that is called when initializing `@primary_key` so that it can use the same logic for each version.

Please review as it may have side effects compared to other changes.